src/JitInterface/src/CorInfoTypes.cs

```diff
@@ -1185,134 +1185,145 @@ public struct NativeVarInfo
         public uint startOffset;
         public uint endOffset;
         public uint varNumber;
-        public VarLoc loc;
-    };
-
-
-    // VarLoc describes the location of a native variable.  Note that currently, VLT_REG_BYREF and VLT_STK_BYREF 
-    // are only used for value types on X64.
-
-    public enum VarLocType
-    {
-        VLT_REG,        // variable is in a register
-        VLT_REG_BYREF,  // address of the variable is in a register
-        VLT_REG_FP,     // variable is in an fp register
-        VLT_STK,        // variable is on the stack (memory addressed relative to the frame-pointer)
-        VLT_STK_BYREF,  // address of the variable is on the stack (memory addressed relative to the frame-pointer)
-        VLT_REG_REG,    // variable lives in two registers
-        VLT_REG_STK,    // variable lives partly in a register and partly on the stack
-        VLT_STK_REG,    // reverse of VLT_REG_STK
-        VLT_STK2,       // variable lives in two slots on the stack
-        VLT_FPSTK,      // variable lives on the floating-point stack
-        VLT_FIXED_VA,   // variable is a fixed argument in a varargs function (relative to VARARGS_HANDLE)
-
-        VLT_COUNT,
-        VLT_INVALID,
+        public VarLoc varLoc;
     };
 
+    // The following 16 bytes come from coreclr types. See comment below.
     public struct VarLoc
     {
-        public VarLocType vlType;
+       public int vlType;
+       // The 64bit field is here to keep VarLoc 8byte aligned
+       public long A;   
```

Comment:
I think this is only true for 64 bit architecture where a pointer size (one of union member) is 8 byte.